### ISSUE TITLE
Shapes Rotation

### ISSUE THREAD

**Issue by Nifriz** [ruby2d/ruby2d#164]
Is it possible implement the same function of [THIS POST](https://github.com/ruby2d/ruby2d/issues/30) to rotate SHAPES?

**Comment by blacktm** [comment]
Hi @Nifriz. Yes, that would be great to add. Not sure I have the math down for rotating arbitrary shapes, but I’m sure we can figure it out (let me know if you have ideas). For zoom, it would be easy for objects that have a width and height — just increase those values keeping the center x, y coordinates fixed. A little trickier for quad and circle. 🤔

**Comment by Nifriz** [comment]
I'm trying a rotation algorithm on Quad Class, I will send code here if it will be a success

**Comment by blacktm** [comment]
Oh, I did write a function to rotate a point around another point, so that could be helpful: https://github.com/simple2d/simple2d/blob/master/src/shapes.c#L6

**Comment by Nifriz** [comment]
Yes very similar to my idea, but I was trying directly on ruby Quad class. Your funcion directly on SDL is certainly more powerful

**Comment by 46bit** [comment]
If anyone comes across this bug needing to do camera movement and zoom, I implemented that for a game of mine. I wrapped around Ruby2D's shapes.

1. Wrap Ruby2D's shapes up so that coordinates and distances can be transformed from virtual to on-screen: https://github.com/46bit/rts/blob/47a9fd0669414fadf93ec7415e969bc3183e119b/lib/renderer/shapes.rb and https://github.com/46bit/rts/blob/47a9fd0669414fadf93ec7415e969bc3183e119b/lib/renderer/ruby2d_shape.rb.
2. Keep track of all resizable shapes and trigger updating them when zooming or panning: https://github.com/46bit/rts/blob/47a9fd0669414fadf93ec7415e969bc3183e119b/lib/renderer/renderer.rb and https://github.com/46bit/rts/blob/47a9fd0669414fadf93ec7415e969bc3183e119b/lib/renderer/camera.rb.

I can't imagine ever contributing this back to Ruby2D, sadly. It's more of a hack than good practice.

**Comment by blacktm** [comment]
No zoom, but rotation works in 9650c7b3c9811bd17d2c6e6205d5660cc9dbf788!
